enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,472 @@
+provide that specific `datacenter.id`
+
+### User Stories
+
+### Risks and Mitigations
+
+- The out-of-tree CCM is required for this work. It will need to be enabled at
+installation time.
+
+## Design Details
+
+The vSphere platform spec and configuration will change to include `vcenters`
+and `regions`. The `MachinePool` will also add an optional `zones` field.
+
+`regions` will contain:
+
+- datacenter
+- vcenter name
+
+`zones` will contain:
+
+- cluster
+- resourcepool
+- datastore
+- network (port group)
+
+which are the required parameters to provision virtual machines across
+datacenters and clusters within a single vcenter.
+
+The vSphere platform spec will be the default configuration for the cloud-config
+adding the additional datacenters from the machinepool.
+
+The master virtual machines will be provisioned with terraform per datacenter
+and cluster. In the case of workers multiple machinesets will need to be
+configured - one per datacenter/cluster pair.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+controlPlane:
+  name: master
+  platform:
+    vsphere:
+      zones:
+      - region-dc1-zone-c1
+      - region-dc1-zone-c2
+      - region-dc1-zone-c3
+      cpus: 8
+      coresPerSocket: 2
+      memoryMB: 24576
+      osDisk:
+        diskSizeGB: 512
+  replicas: 3
+compute:
+- name: worker
+  platform:
+    vsphere:
+      zones:
+      - region-dc1-zone-c1
+      - region-dc1-zone-c2
+      - region-dc1-zone-c3
+      cpus: 8
+      coresPerSocket: 2
+      memoryMB: 24576
+      osDisk:
+        diskSizeGB: 512
+  replicas: 3
+metadata:
+  name: test-cluster
+platform:
+  vSphere:
+    vCenter: your.vcenter.example.com
+    username: username
+    password: password
+    datacenter: datacenter
+    defaultDatastore: datastore
+    network: portgroup
+    vcenters:
+    - name: your.first-vcenter.example.com
+      username: username
+      password: password
+    - name: your.second-vcenter.example.com
+      username: username
+      password: password
+    regions:
+    - name: region-dc1
+      datacenter: dc1
+      vcenter: your.first-vcenter.example.com
+      zones:
+      - name: region-dc1-zone-c1
+        cluster: DC1_C1
+        resourcepool: /DC1/host/DC1_C1/Resources
+        network: "vm network"
+        datastore: "DC1_DS1"
+      - name: region-dc1-zone-c2
+        cluster: DC1_C2
+        resourcepool: /DC1/host/DC1_C2/Resources
+        network: "vm network"
+        datastore: "DC1_DS2"
+      - name: region-dc1-zone-c3
+        cluster: DC1_C3
+        resourcepool: /DC1/host/DC1_C3/Resources
+        network: "vm network"
+        datastore: "DC1_DS3"
+pullSecret:
+sshKey:
+```
+
+Each vcenter datacenter defined in either master or worker machinepool will
+need to be added to the out-of-tree CCM configuration. This allows the CCM
+to find the virtual machines across multiple vcenter datacenters.
+
+### Open Questions
+
+- What is the expected latency between vCenter Clusters and RHCOS virtual
```

Comment:
Why does vCenter have to talk to etcd? Maybe the question needs to be clarified 🤔